tests/parse_self.rs

```diff
@@ -90,6 +90,18 @@ fn test_parse_self_debug_line() {
             while let Some(_) = rows.next_row()
                 .expect("Should parse and execute all rows in the line number program") {
             }
+
+            let header = debug_line.header(offset, unit.address_size(), comp_dir, comp_name)
+                .expect("should parse line number program header");
+            let (header, sequences) = header.sequences()
+                .expect("should parse and execute the entire line number program");
+            assert!(sequences.len() > 0); // Should be at least one sequence.
+            for sequence in sequences {
+                let mut resumed = header.clone().resume_from(&sequence);
```

Comment:
It would be nice to avoid needing to clone the header to iterate the rows in a sequence. So the caller would need to somehow obtain a `StateMachine` that already has a `Complete` header, and resume on that instead.